patches/minecraft/net/minecraft/client/Minecraft.java.patch

```diff
@@ -153,15 +170,26 @@
                    this.field_71442_b.func_180511_b(blockpos, blockraytraceresult.func_216354_b());
                    break;
                 }
-@@ -1187,6 +1196,7 @@
+@@ -1187,8 +1206,10 @@
                 }
  
                 this.field_71439_g.func_184821_cY();
 +               net.minecraftforge.common.ForgeHooks.onEmptyLeftClick(this.field_71439_g);
              }
  
++            if (inputEvent.shouldSwingHand())
              this.field_71439_g.func_184609_a(Hand.MAIN_HAND);
-@@ -1236,6 +1246,9 @@
+          }
+       }
+@@ -1203,6 +1224,7 @@
+             }
+ 
+             for(Hand hand : Hand.values()) {
++               if (net.minecraftforge.client.ForgeHooksClient.onClickInput(1, this.field_71474_y.field_74313_G, hand).isCanceled()) continue;
```

Comment:
I would suggest you "return" instead of "continue" when a hand interaction is cancelled as cancelling should mean that it was a successful action. This will match how Minecraft handles successful interactions. Just look at the code below for how it handles it with ActionResultType.

Reply:
I originally had it as return, but I switched it to continue because it represents a ActionResultType.PASS. If you want to block the main hand from doing something you might still want the off hand to have default behaviour. If users want it to function like return they can always add a ``if(getHand()==OFF_HAND) return`` themselves.

Reply:
It would be better to represent cancelled as successful. You don't have to write extra code. Modders should be cancelling the event inside of a condition.

Reply:
Your behaviour simply makes the code verbose for no reason. My answer has the exact same behaviour, you don't need to write `if(getHand()==OFF_HAND) return` at all.